pandas/core/indexes/datetimelike.py

```diff
@@ -855,3 +857,48 @@ def f(self):
     f.__name__ = fget.__name__
     f.__doc__ = fget.__doc__
     return property(f)
+
+
+class DatetimelikeDelegateMixin(PandasDelegate):
```

Comment:
Is PandasDelegate overkill for this situation?  It's not like `Series.dt` where we construct an object and need to cache it.  Couldn't we just use something like:

```
def pass_through_methods(names):
    def decorator(cls):
        for name in names:
            method = wrap_array_method(...)
            setattr(cls, name, method)
        return cls
```

There's definitely overlap with the PandasDelegate machinery, but I think that machinery makes for really dense reading, so should be reserved for when we really need it.

Reply:
Perhaps you're thinking of `Properties` (which subclasses `PandasDelegate`)? `PandasDelegate` doesn't have an `__init__`, it just handles registering and creating the delegated properties.

Reply:
But, I don't have a strong preference of one of the other. We use `PandasDelegate` on master, for period, so this PR is a strict generalization of that.